make/autoconf/jdk-version.m4

```diff
@@ -480,91 +490,92 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
-    if test "x$VERSION_PRE" != x; then
-      MACOSX_BUNDLE_ID_BASE="$MACOSX_BUNDLE_ID_BASE-$VERSION_PRE"
-    fi
+  # If using the default value, append the VERSION_PRE if there is one
+  # to make it possible to tell official builds apart from developer builds
+  if test "x$VERSION_PRE" != x; then
+    MACOSX_BUNDLE_ID_BASE="$MACOSX_BUNDLE_ID_BASE-$VERSION_PRE"
   fi
+
+  # Set the MACOSX Bundle ID base
+  UTIL_ARG_WITH(NAME: macosx-bundle-id-base, TYPE: string,
+    DEFAULT: $MACOSX_BUNDLE_ID_BASE,
+    DESC: [Set the MacOSX Bundle ID base. This is the base ID for calculating MacOSX Bundle IDs.],
+    DEFAULT_DESC: [not specified],
+    CHECK_VALUE: [
+      if test "x$RESULT" = x; then
+        FAILURE="--with-macosx-bundle-id-base must have a value"
+      elif [ ! [[ $RESULT =~ ^[[:print:]]*$ ]] ]; then
+        FAILURE="--with-macosx-bundle-id-base contains non-printing characters: $RESULT"
+      fi
+    ])
   AC_SUBST(MACOSX_BUNDLE_ID_BASE)
 
-  # Set the MACOSX CFBundleVersion field
-  AC_ARG_WITH(macosx-bundle-build-version, [AS_HELP_STRING([--with-macosx-bundle-build-version],
-      [Set the MacOSX Bundle CFBundleVersion field. This key is a machine-readable
-      string composed of one to three period-separated integers and should represent the
-      build version. Defaults to the build number.])])
-  if test "x$with_macosx_bundle_build_version" = xyes || test "x$with_macosx_bundle_build_version" = xno; then
-    AC_MSG_ERROR([--with-macosx-bundle-build-version must have a value])
-  elif [ ! [[ $with_macosx_bundle_build_version =~ ^[0-9\.]*$ ]] ]; then
-    AC_MSG_ERROR([--with-macosx-bundle-build-version contains non numbers and periods: $with_macosx_bundle_build_version])
-  elif test "x$with_macosx_bundle_build_version" != x; then
-    MACOSX_BUNDLE_BUILD_VERSION="$with_macosx_bundle_build_version"
+  if test "x$VERSION_BUILD" != x; then
+    MACOSX_BUNDLE_BUILD_VERSION="$VERSION_BUILD"
   else
-    if test "x$VERSION_BUILD" != x; then
-      MACOSX_BUNDLE_BUILD_VERSION="$VERSION_BUILD"
-    else
-      MACOSX_BUNDLE_BUILD_VERSION=0
-    fi
+    MACOSX_BUNDLE_BUILD_VERSION=0
+  fi
 
-    # If VERSION_OPT consists of only numbers and periods, add it.
-    if [ [[ $VERSION_OPT =~ ^[0-9\.]+$ ]] ]; then
-      MACOSX_BUNDLE_BUILD_VERSION="$MACOSX_BUNDLE_BUILD_VERSION.$VERSION_OPT"
-    fi
+  # If VERSION_OPT consists of only numbers and periods, add it.
+  if [ [[ $VERSION_OPT =~ ^[0-9\.]+$ ]] ]; then
+    MACOSX_BUNDLE_BUILD_VERSION="$MACOSX_BUNDLE_BUILD_VERSION.$VERSION_OPT"
   fi
+
+  # Set the MACOSX CFBundleVersion field
+  UTIL_ARG_WITH(NAME: macosx-bundle-build-version, TYPE: string,
+    DEFAULT: $MACOSX_BUNDLE_BUILD_VERSION,
+    DESC: [Set the MacOSX Bundle CFBundleVersion field. This key is a machine-readable
+      string composed of one to three period-separated integers and should represent the
+      build version. Defaults to the build number.],
```

Comment:
Please remove the text about the default from DESC, and add:
```
DEFAULT_DESC: [the build number],
```

Reply:
Done, thanks